### PR TITLE
graphql, indexer: introduce ephemeral postgres db for testing

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Convert all tests that relied on an external postgres db to use an ephemeral pstrgres db enabling them all to be run in parallel.

After this patch the graphql-test workflow takes ~7 minutes compared to ~19 minutes it used to take.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-graphql-rpc/src/test_infra/cluster.rs [hunk 1]
```diff
@@ -196,9 +215,45 @@ pub async fn serve_executor(
...
+    sim.create_checkpoint();
+    sim.create_checkpoint();
+    sim.create_checkpoint();
+    sim.advance_epoch(true);
+    sim.create_checkpoint();
+    sim.advance_clock(
+        std::time::SystemTime::now()
+            .duration_since(std::time::SystemTime::UNIX_EPOCH)
+            .unwrap(),
+    );
```
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/test_infra/cluster.rs:238]: should we advance clock for each checkpoint to better emulate what happens in prod

File: crates/sui-indexer/src/tempdb.rs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        Ok(())
+    } else {
+        Err(anyhow!("unable to initialize database"))
+    }
+}
+
+/// Return an ephemeral, available port. On unix systems, the port returned will be in the
+/// TIME_WAIT state ensuring that the OS won't hand out this port for some grace period.
+/// Callers should be able to bind to this port given they use SO_REUSEADDR.
+pub fn get_available_port() -> u16 {
```
Comment (wlmyng) [inline crates/sui-indexer/src/tempdb.rs:290]: throughout our tests we have a few places that do `("127.0.0.1:{}", get_available_port)`, what are your thoughts on having a function here that's like, `build_db_address(host: Option<u64>)` so we can reduce the manual `127.0.0.1`

**Review by wlmyng — APPROVED** [review]
dope

File: crates/sui-graphql-rpc/src/test_infra/cluster.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-graphql-rpc/src/test_infra/cluster.rs:238]: this is specifically to get the clock to be close to "current time" for the health check test. there's probably some changes we could make to Simulacrum to make this a bit easier of course

**Pull request merged**